src/server/scripts/Spells/spell_druid.cpp

```diff
@@ -1486,6 +1495,136 @@ class spell_dru_wild_growth : public SpellScriptLoader
     }
 };
 
+// 79577 - Eclipse
+class spell_dru_eclipse_dummy : public AuraScript
+{
+    PrepareAuraScript(spell_dru_eclipse_dummy);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_DRUID_BALANCE_STARFIRE, SPELL_DRUID_BASE_STARFIRE,
+            SPELL_DRUID_BALANCE_WRAITH, SPELL_DRUID_BASE_WRAITH });
+    }
+
+    bool Load() override
+    {
+        return GetCaster()->GetTypeId() == TYPEID_PLAYER;
+    }
+
+    void HandleProc(ProcEventInfo& eventInfo)
+    {
+        if (SpellInfo const* spellInfo = eventInfo.GetSpellInfo())
+        {
+            if (Player* player = GetCaster()->ToPlayer())
+            {
+                bool isBalance = player->GetPrimarySpecialization() == TALENT_SPEC_DRUID_BALANCE;
+                uint32 wraith = isBalance ? SPELL_DRUID_BALANCE_WRAITH : SPELL_DRUID_BASE_WRAITH;
+                uint32 starfire = isBalance ? SPELL_DRUID_BALANCE_STARFIRE : SPELL_DRUID_BASE_STARFIRE;
+                if (spellInfo->Id == starfire)
+                    OnSpellCast(SPELL_DRUID_ECLIPSE_SOLAR_SPELL_CNT, SPELL_DRUID_ECLIPSE_LUNAR_SPELL_CNT, SPELL_DRUID_ECLIPSE_SOLAR_AURA);
+                else if (spellInfo->Id == wraith)
+                    OnSpellCast(SPELL_DRUID_ECLIPSE_LUNAR_SPELL_CNT, SPELL_DRUID_ECLIPSE_SOLAR_SPELL_CNT, SPELL_DRUID_ECLIPSE_LUNAR_AURA);
+            }
+        }
+    }
+
+    void HandleApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        SetSpellCount(SPELL_DRUID_ECLIPSE_SOLAR_SPELL_CNT, 2);
```

Comment:
Replace hardcoded 2 with `aurEff->GetAmount()` - there are auras that affect eclipse counting system (334604) and they modify the value with a SPELLMOD_EFFECT1 (already working, you don't need to do anything)